tests/test_sshfs.py

```diff
@@ -71,6 +72,29 @@ def strip_keys(info):
         info.pop(key, None)
 
 
+def test_fsspec_registration(fs, ssh_server, remote_dir, user="user"):
+    """Check that we can use sshfs with fsspec.open and an ssh url."""
+    path = remote_dir + "/a.txt"
+
+    # Additional options like client_keys need to be passed to open.
+    url = f"ssh://{user}@{ssh_server.host}:{ssh_server.port}/{path}"
+    with fsspec.open(url, "w", client_keys=[USERS[user]]) as file:
+        # Check the underlying file system.
+        file_fs = file.buffer.fs
+        assert isinstance(file_fs, SSHFileSystem)
+        assert file_fs.storage_options == {
+            "host": ssh_server.host,
+            "port": ssh_server.port,
+            "username": user,
+            "client_keys": [USERS[user]],
+        }
+        # Write something.
+        file.write("Anything")
+
+    with fs.open(path, "r") as file:
+        assert file.read() == "Anything"
```

Comment:
I suppose you could just do something like:

```
fs = fsspec.filesystem("ssh")
assert isinstance(fs, SSHFileSystem)
```

The rest of open/read/write testing is redundant and is out of scope for testing if sshfs is plugged into fsspec correctly.

Reply:
I think it's worth testing that the URL is parsed correctly, i.e. user/port/hostname.

But I see that the write and read are redundant, I'll remove them.

I'll also add your suggestion as a separate test, i.e. to have:
- one test for registration (just testing `filesystem("ssh")`)
- one for URL parsing

The best way I have found for URL parsing is to use `fsspec.open`. Maybe there is an easier way? (Similar to `fsspec.filesystem`? If you know one, please let me know, because I couldn't find it.

Reply:
Makes sense 👍 Thank you!

Reply:
Done. I realized we cannot just do `fsspec.filesystem("ssh")`, as SSHFileSystem tries to connect immediately and fails if it can't. So we need to provide host etc. to that call.

Still, I think the tests are clearer this way.

Not sure if the `open` route is the best way to test URL parsing, but as mentioned, the best one I found so far.